flink-test-utils-parent/flink-test-utils/src/main/java/org/apache/flink/test/util/MiniClusterResource.java

```diff
@@ -69,6 +73,10 @@
 
 	private int numberSlots = -1;
 
+	private Collection<Path> jarFiles = Collections.emptyList();
+
+	private Collection<URL> classPaths = Collections.emptyList();
```

Comment:
I think the `jarFiles` and the `classPaths` should not be part of the `MiniClusterResource`, because they belong to the client context.

Reply:
Yes I wasn't so sure about this either. But we either override do it like this or override part of the global state that the `MiniClusterResource` modifies when it sets the context :/

Reply:
Then maybe, let's not use the `MiniClusterResource` directly but instantiate a `MiniCluster` instead.

Reply:
that make sense

Reply:
We then have to duplicate the test though.

Reply:
I would be ok with that given that we will remove the duplicate quite soonish when we remove the legacy code.